vllm/v1/worker/gpu_model_runner.py

```diff
@@ -389,6 +392,18 @@ def _update_states(self, scheduler_output: "SchedulerOutput") -> None:
                     req_index, start_index:end_token_index] = spec_token_ids
             # NOTE(woosuk): `num_tokens` here may include spec decode tokens.
             self.input_batch.num_tokens[req_index] = end_token_index
+            # Fill the bitmask
+            if (req_id in scheduler_output.guided_decoding_request_ids
+                    and req_state.grammar is not None):
+                if not req_state.grammar.prefilled:
+                    req_state.grammar.prefilled = True
+                else:
+                    token_idx = scheduler_output.num_scheduled_tokens[
+                        req_id] - 1
+                    if not req_state.grammar.matcher.is_terminated():
+                        assert req_state.grammar_bitmask is not None
+                        req_state.grammar.fill_bitmask(
+                            req_state.grammar_bitmask, token_idx)
```

Comment:
Does token_idx always 1 here? I think currently the batch dimension of the bitmask is always 1